### ISSUE TITLE
question about configuration of ldap through pam

### ISSUE THREAD

**Issue by yr-wang-hit** [irods/irods#6278]
We aim to authenticate irods access through ldap in our center.
we configure irods SSL/pam/ldap with the help of online official doc and slides from 
`` https://slides.com/irods/irods-ldap ``
`` https://slides.com/irods/irods-pam `` 
But it cannot pass the test on 
`` /usr/sbin/irodsPamCheck user ``
`` password ``
`` Not authenticated `` 
We want to consult how to confirm if the pam configuration and ldap-pam connection is successfully established.
It's really confusing that even the ldap admin account cannot pass the test. If there are some logs or indicating messages to 
check where the misconfiguration occupied.

Thanks a lot.

**Comment by korydraughn** [comment]
Q. What version of iRODS are you running?
Q. Have you checked the rodsLog? (located in `/var/lib/irods/log` if installed via packages)

**Comment by yr-wang-hit** [comment]
Sorry for late response.
Our situation is that ` ldapsearch ` can successfully retrieve the information of our ldap server.
But the ` /usr/sbin/irodsPamAuthCheck ` still report ` Not AUTHENTICATED `

The log report the CS_NEG_FAILURE and looks like:


`  ::1 ERROR: [-] /repos/irods/server/core/src/rodsAgent.cpp:521:int runIrodsAgentFactory(sockaddr_un) :  status [SERVER_NEGOTIATION_ERROR]  errno [] -- message [failure detected from client for result [CS_NEG_FAILURE]] ` 

Thanks a lot!

**Comment by trel** [comment]
Make sure you try this part of the documentation...
 - https://docs.irods.org/4.2.11/plugins/pluggable_authentication/#pam-pluggable-authentication-module

> If irodsPamAuthCheck returns `Not Authenticated`, that suggests that PAM is not set up correctly. You will need to configure PAM correctly (and therefore get irodsPamAuthCheck returning Authenticated) before using PAM through iRODS.
> 
> A simple way to check that you are using irodsPamAuthCheck correctly, and that it is the PAM settings that need updated, is to create a fully permissive PAM setup with the following command.
> 
> ```
> sudo su - root -c 'echo "auth sufficient pam_permit.so" > /etc/pam.d/irods'
> ```
> This will allow any username/password combination to successfully authenticate with the irods PAM service, meaning that any username/password combination should cause irodsPamAuthCheck to return `Authenticated`.
> 
> With the permissive configuration working with irodsPamAuthCheck, the next step is to adjust your PAM configuration to your desired settings (LDAP, in this case). You will know that is correct when irodsPamAuthCheck behaves as you would expect when using LDAP username/passwords. iRODS uses irodsPamAuthCheck directly, so if it is working on the command line, it should work when run by iRODS.

**Comment by yr-wang-hit** [comment]
Thanks Trel!
I have tried this part of documentation and it will pass the test if we create the fully permissive PAM setup, Any combination of 
username and password will pass the check. 
After that, we may change the content of /etc/pam.d/irods to something like 
`  auth sufficient pam_ldap.so `
isn't it?

And it cannot pass the check if we tape in username and password in our ldap server this turn.

**Comment by trel** [comment]
Yes, if you can get the `/etc/pam.d/irods` file to work, you're good to go.

Here are the pam_ldap docs:  https://linux.die.net/man/5/pam_ldap

**Comment by yr-wang-hit** [comment]
Thanks trel, Everything goes well now and I will close the issue.

**Comment by trel** [comment]
Excellent!   Can you share your working `/etc/pam.d/irods` file?

As long as there are no secret things in there, of course.

**Comment by yr-wang-hit** [comment]
The only line is:
` auth sufficient pam_ldap.so `
It is a test setup and we will try to add other authentication rules if needed.

Thanks.
